tensorflow-keras/src/main/java/org/tensorflow/keras/optimizers/OptimizerInterface.java

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.keras.optimizers;
+
+import java.util.Map;
+import org.tensorflow.Graph;
+import org.tensorflow.op.Ops;
+
+/** The main Interface for Keras Optimizers */
+public interface OptimizerInterface {
```

Comment:
If possible, I would prefer we avoid to suffix this class name with `Interface`, since none of our other interfaces have this suffix.

Reply:
I was trying to avoid a name clash with `org.tensorflow.framework.optimizers.Optimizer`. Some of OptimizerInterface could be refactored into org.tensorflow.framework.optimizers.Optimizer. At the time, I was trying to keep the two projects, keras and frameworks, totally separate.

Reply:
We could call this `KerasOptimizer`? The fact that python overloads all the names doesn't mean we need to. Or rename `org.tensorflow.framework.optimizers.Optimizer` to `BaseOptimizer` or `FrameworkOptimizer`?

Reply:
Actually, I have done away with the `KerasOptimizer` interface for now and put `assertGraph` as a static call in a helper class. However, this helper class will probably be useful in some of the other Keras packages that use `Variables`, so maybe it should be moved to a general util package that is only visible to the java module, wherever we decide to put it.

One question that remains for the `Optimizer` interface, is where to put the method prototypes for `get/setLearningRate`, when that feature comes in the next PR. The Keras code will need to treat these methods as a general `Optimizer`, rather than as individual types of `Optimizer`. There are Optimizers that I found on the Web that don't use learning rate, but all of the currently defined Optimizers in the framework do use it.